### ISSUE TITLE
When using full paths to source xaml files, the generated .g.cs files aren't placed in subfolders in the obj\target folder

### ISSUE THREAD

**Issue by emusgrave** [dotnet/wpf#6290]

**Comment by lindexi** [comment]
@emusgrave Because the XAML Compiler do not know the relative path and it do not know where is the project root folder.

**Comment by lindexi** [comment]
And I think few developers will write absolute paths in projects.

**Comment by emusgrave** [comment]
@lindexi I agree that not many will use absolute paths, because they would have to be writing the csproj by hand. Or, in this case, using `cmake` to generate it. But _anyone_ using cmake and also having duplicate file names with encounter this. Cmake is designed to always support out-of-source builds, so there is no way around the absolute paths.

My workaround was to guarantee that each file has a unique name, so they don't conflict when they all get written to the same output folder. 

Perhaps an error message could at least be generated by the compiler when it encounters a duplicate file name during a single build run (as it tries to write the output)?

**Comment by SeSamSeg** [comment]
I run into this with the following setup:
Shared\SomeFolder\SomePage.xaml
Shared\SomeOtherFolder\SomePage.xaml

### Shared\Shared.props
```
  <PropertyGroup>
    <DefaultItemExcludes>$(DefaultItemExcludes);$(MSBuildThisFileDirectory)obj\/**;$(MSBuildThisFileDirectory)bin\/**</DefaultItemExcludes>
    <DefaultExcludesInProjectFolder>$(DefaultItemExcludes);$(MSBuildThisFileDirectory)**/.*/**;$(MSBuildThisFileDirectory)**/*.user</DefaultExcludesInProjectFolder>
  </PropertyGroup>
  <ItemGroup>
    <Compile
      Include="$(MSBuildThisFileDirectory)**/*$(DefaultLanguageSourceExtension)"
      Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)"
    />
    <EmbeddedResource
      Include="$(MSBuildThisFileDirectory)**/*.resx"
      Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)"
    />
    <Page
      Include="$(MSBuildThisFileDirectory)**/*.xaml"
      Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);App.xaml;Application.xaml"
    />
  </ItemGroup>
  <ItemGroup>
    <None
      Include="$(MSBuildThisFileDirectory)**/*"
      Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)"
    />
    <None Remove="$(MSBuildThisFileDirectory)**/*$(DefaultLanguageSourceExtension)" />
    <None Remove="$(MSBuildThisFileDirectory)**/*.resx" />
    <None Remove="$(MSBuildThisFileDirectory)**/*.xaml" />
  </ItemGroup>
```

### Target1\Target1.csproj
```
  <ItemGroup>
    <ProjectReference Include="..\Shared\Shared.props" />
  </ItemGroup>
```
I have multiple projects that share a lot of files that are build with different dependencies and defines, but most of the content is the same.
Of course it would be best to split off in libraries and abstract the changes, but this is my current setup and the name conflict was not obvious as the error was the following which took some time to link to a name confict:
 > Duplicate 'Compile' items were included. The .NET SDK includes 'Compile' items from your project directory by default. You can either remove these items from your project file, or set the 'EnableDefaultCompileItems' property to 'false' if you want to explicitly include them in your project file. For more information, see https://aka.ms/sdkimplicititems. The duplicate items were: 'C:\...\Target1\obj\Debug\net8.0-windows\win-x64\SomePage.g.i.cs'
